src/main/host/syscall/socket.c

```diff
@@ -991,26 +993,41 @@ SysCallReturn syscallhandler_getsockopt(SysCallHandler* sys,
     }
     utility_assert(socket_desc);
 
-    /* The pointers must be non-null. */
-    if (!optvalPtr.val || !optlenPtr.val) {
+    /* The optlen pointer must be non-null. */
+    if (!optlenPtr.val) {
         return (SysCallReturn){.state = SYSCALL_DONE, .retval.as_i64 = -EFAULT};
     }
 
+    /* Get the len via clone, so we can write to optlenPtr too. */
```

Comment:
I know you just moved this, but could you update this stale comment? I think we can just drop it.

We used to use a "clone" API here as a workaround for being able to read + write the same pointer, but don't need it since we added `process_getMutablePtr`.